pkg/internal/discover/attacher_linux.go

```diff
@@ -46,6 +48,12 @@ func (ta *TraceAttacher) bpfMount(pinPath string) error {
 		return err
 	}
 	if !mounted {
+		caps, err := helpers.GetCurrentProcCapabilities()
+
+		if err == nil && !caps.Has(unix.CAP_SYS_ADMIN) {
+			return fmt.Errorf("beyla requires CAP_SYS_ADMIN in order to mount %s", pinPath)
```

Comment:
Just wondering if there is any possibility that the FS is not mounted by another reason than not having CAP_SYS_ADMIN? In that case the error would be misleading.

Reply:
It could be, but `CAP_SYS_ADMIN` is a prerequisite to mounting anyway, so without it, it will invariably fail. If we _do_ have `CAP_SYS_ADMIN`, but there's another reason that is preventing us from mounting, then that reason will error out in the next statement `unix.Mount()`.